storage-mirror-firestore/extension.yaml

```
@@ -0,0 +1,193 @@
+      Firestore appropriately.
+    properties:
+      location: ${LOCATION}
+      runtime: nodejs10
+      eventTrigger:
+        eventType: google.storage.object.finalize
+        resource: projects/_/buckets/${BUCKET}
+  - name: mirrorDelete
+    type: firebaseextensions.v1beta.function
+    description:
+      Listens for changes in your specified Cloud Storage bucket, and updates
+      Firestore appropriately.
+    properties:
+      location: ${LOCATION}
+      runtime: nodejs10
+      eventTrigger:
+        eventType: google.storage.object.delete
+        resource: projects/_/buckets/${BUCKET}
+  - name: mirrorArchive
+    type: firebaseextensions.v1beta.function
+    description:
+      Listens for changes in your specified Cloud Storage bucket, and updates
+      Firestore appropriately.
+    properties:
+      location: ${LOCATION}
+      runtime: nodejs10
+      eventTrigger:
+        eventType: google.storage.object.archive
+        resource: projects/_/buckets/${BUCKET}
+  - name: mirrorMetadataUpdate
+    type: firebaseextensions.v1beta.function
+    description:
+      Listens for changes in your specified Cloud Storage bucket, and updates
+      Firestore appropriately.
+    properties:
+      location: ${LOCATION}
+      runtime: nodejs10
+      eventTrigger:
+        eventType: google.storage.object.metadataUpdate
+        resource: projects/_/buckets/${BUCKET}
+
+params:
+  - param: LOCATION
+    type: select
+    label: Cloud Functions location
+    description: >-
+      Where do you want to deploy the functions created for this extension?
+      You usually want a location close to your database. For help selecting a
+      location, refer to the [location selection
+      guide](https://firebase.google.com/docs/functions/locations).
+    options:
+      - label: Iowa (us-central1)
+        value: us-central1
+      - label: South Carolina (us-east1)
+        value: us-east1
+      - label: Northern Virginia (us-east4)
+        value: us-east4
+      - label: Belgium (europe-west1)
+        value: europe-west1
+      - label: London (europe-west2)
+        value: europe-west2
+      - label: Hong Kong (asia-east2)
+        value: asia-east2
+      - label: Tokyo (asia-northeast1)
+        value: asia-northeast1
+    default: us-central1
+    required: true
+    immutable: true
+
+  - param: BUCKET
+    type: string
+    label: Cloud Storage bucket to mirror
+    description: >
+      Which Cloud Storage bucket should we mirror?
+    default: ${STORAGE_BUCKET}
+    # validationRegex: (.)+\.(.)+\.(.)+
+    validationErrorMessage: Invalid storage bucket
+    example: my-project-12345.appspot.com
+    required: true
+
+  - param: FIRESTORE_ROOT
+    label: Root Document path
+    description: >
+      What is the path to the collection where documents references should be
+      stored?
+    validationRegex: "^[^/]+/[^/]+(/[^/]+/[^/]+)*$" # Must be a path to a document.
```

Comment is covered by the description
```suggestion
    validationRegex: "^[^/]+/[^/]+(/[^/]+/[^/]+)*$"
```